include/deal.II/lac/solver_cg.h

```diff
@@ -265,16 +269,17 @@ SolverCG<VectorType>::print_vectors(const unsigned int,
 template <typename VectorType>
 inline void
 SolverCG<VectorType>::compute_eigs_and_cond(
-  const std::vector<double> &diagonal,
-  const std::vector<double> &offdiagonal,
+  const std::vector<typename VectorType::value_type> &diagonal,
+  const std::vector<typename VectorType::value_type> &offdiagonal,
```

Comment:
(above and below).

... we have to make `solver_cg` (that is used for one of the boundary variants) compatible with complex numbers. I have fixed compilation but this isn't tested for functionality yet and misses complex conjugation for the scalar product. But that should be an easy fix.